features/matchers/send_email_matcher.feature

```diff
@@ -0,0 +1,85 @@
+Feature: `send_email` matcher
+
+  The `send_email` matcher is used to check if an email with the given parameters has been sent during a spec example run.
+
+  NOTE: It implies that the spec example actually sends the email using the test adapter and does not schedule it for background.
+  To have an email sent in tests make sure:
+  - `ActionMailer` performs deliveries - `Rails.application.config.action_mailer.perform_deliveries = true`
+  - If the email is sent asynchronously (with `.deliver_later` call), ActiveJob uses the inline adapter - `Rails.application.config.active_job.queue_adapter = :inline`
+  - ActionMailer uses the test adapter - `Rails.application.config.action_mailer.delivery_method = :test`
+  If you want to check an email has been scheduled for background, use the `have_enqueued_email` matcher.
```

Comment:
These get rendered as part of our docs when published, so can you break this up for readability?

```suggestion
  NOTE: It implies that the spec example actually sends the email using the test adapter and does not schedule it for background.
  
  To have an email sent in tests make sure:
  - `ActionMailer` performs deliveries - `Rails.application.config.action_mailer.perform_deliveries = true`
  - If the email is sent asynchronously (with `.deliver_later` call), ActiveJob uses the inline adapter - `Rails.application.config.active_job.queue_adapter = :inline`
  - ActionMailer uses the test adapter - `Rails.application.config.action_mailer.delivery_method = :test`
  
  If you want to check an email has been scheduled for background, use the `have_enqueued_email` matcher.
```